src/plugins/customdata.ts

```diff
@@ -0,0 +1,28 @@
+// Allows data sent via. normal DOM event handlers to be logged alongside Clarity data by listening to the
+// custom DOM event "claritydata".
+import { addEvent, bind } from "./../core";
+
+// Events coming from normal DOM will have additional data in details property
+interface ICustomEvent extends Event {
+    detail: any;
+}
+
+export default class CustomData implements IPlugin {
+    private eventName = "Custom";
+
+    public activate() {
+        bind(window, "claritydata", this.customDataHandler.bind(this));
+    }
+
+    public reset(): void {
+        return;
+    }
+
+    public teardown() {
```

Comment:
Had a quick conversation with Toby 7/21. Since current Clarity modules (including pointer and errors) do not unbind handlers on cleanup, this behavior is consistent. Should consider a future conversation about whether Clarity is ever expected to be unloaded independent of page unload.